dbms/src/Functions/FunctionsStringReplace.h

```diff
@@ -280,7 +232,28 @@ class FunctionStringReplace : public IFunction
             const auto * col_replacement_const = typeid_cast<const ColumnConst *>(column_replacement.get());
             auto replacement = col_replacement_const->getValue<String>();
 
-            if (const auto * col = checkAndGetColumn<ColumnString>(column_src.get()))
+            bool col_const = column_src->isColumnConst();
+
+            if (col_const)
+            {
+                auto new_src = column_src->convertToFullColumnIfConst();
```

Comment:
This cost is too high, why not simply provide a reference?

Reply:
here i use:
```cpp
// using the data directly as a reference.
const auto & const_data = col_const->getDataColumn();
const auto * col = typeid_cast<const ColumnString *>(&const_data);
```
instad to avoid these cost now.